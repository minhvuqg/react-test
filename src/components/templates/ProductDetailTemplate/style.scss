.quantity {
  width: 30%;
  padding-left: 64px;

  .quantity-card {
    width: 200px;
    height: 200px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;

    .quantity-buttons {
      display: flex;
      justify-content: space-between;
      width: 100%;
      align-items: center;
    }

    .add-to-cart-btn {
      width: 100%;
      height: 40px;
      cursor: pointer;
    }
  }
}
