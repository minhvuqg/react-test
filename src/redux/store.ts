import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'

import productSlice from './reducers/productSlice'

const customizedMiddleware: any = getDefaultMiddleware({
  serializableCheck: false
})

const persistedReducer: any = combineReducers({
  cart: productSlice
})

export default configureStore({
  reducer: persistedReducer,
  middleware: customizedMiddleware
})
