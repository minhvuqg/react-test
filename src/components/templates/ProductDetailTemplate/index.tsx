import { FC, useEffect, useState } from 'react'

import { useDispatch } from 'react-redux'
import { useParams } from 'react-router-dom'

import { ProductQuantity } from 'components/UI/molecules/ProductQuantity'
import { ProductDetails } from 'components/UI/molecules/ProductDetails'
import { Button } from 'components/UI/atoms/Button'

import { addToCart } from 'redux/reducers/productSlice'
import { productList } from 'data'

import { IProps } from './type.d'

import './style.scss'

const ProductDetailTemplate: FC<IProps> = (): JSX.Element => {
  const params: { id: string } = useParams()
  const dispatch: any = useDispatch()

  const [product, setProduct] = useState<Product.Infor | undefined>()
  const [quantity, setQuantity] = useState<number>(1)

  useEffect((): void => {
    const productTemp: Product.Infor | undefined = productList.find(
      (productItem: Product.Infor): boolean => productItem.id === params.id
    )

    setProduct(productTemp)
  }, [params.id])

  const incrementQuantity: any = (): void => {
    setQuantity(quantity + 1)
  }

  const decrementQuantity: any = (): void => {
    if (quantity > 1) {
      setQuantity(quantity - 1)
    }
  }

  const handleAddToCart: any = (): void => {
    dispatch(addToCart({ ...product, quantity }))
  }

  if (!product) {
    return <div>Not found</div>
  }

  return (
    <>
      <ProductDetails product={product} />

      <div className="quantity">
        <div className="quantity-card">
          <div className="quantity-buttons">
            <span>Quantity</span>
            <ProductQuantity
              quantity={quantity}
              incrementQuantity={incrementQuantity}
              decrementQuantity={decrementQuantity}
            />
          </div>

          <Button
            primary={true}
            className="add-to-cart-btn"
            onClick={handleAddToCart}
          >
            Add to cart
          </Button>
        </div>
      </div>
    </>
  )
}

export default ProductDetailTemplate
