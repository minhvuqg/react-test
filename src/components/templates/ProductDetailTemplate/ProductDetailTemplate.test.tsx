import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import ProductDetailTemplate from '../ProductDetailTemplate'
import store from 'redux/store'
import { BrowserRouter as Router } from 'react-router-dom'

it('render productTemp correctly', () => {
  const productTempEl = TestRenderer.create(
    < Provider store={store} >
      <Router>
        <ProductDetailTemplate />
      </Router>
    </Provider >).toJSON()
  expect(productTempEl).toMatchSnapshot()
})
