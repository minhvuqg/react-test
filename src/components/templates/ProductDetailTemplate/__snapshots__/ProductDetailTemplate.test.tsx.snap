// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render productTemp correctly 1`] = `
<div>
  Not found
</div>
`;
